ts/features/wallet/onboarding/bancomat/screens/SearchBankScreen.tsx

```diff
@@ -1,18 +1,137 @@
+import { withLightModalContext } from "../../../../../components/helpers/withLightModalContext";
+import { LightModalContextInterface } from "../../../../../components/ui/LightModal";
+import TosBonusComponent from "../../../../bonus/bonusVacanze/components/TosBonusComponent";
 
+type Props = LightModalContextInterface &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
 /**
  * This screen allows the user to choose a specific bank to search for their Bancomat.
  * the user can also choose not to specify any bank and search for all Bancomat in his name
  * @constructor
  */
-export const SearchBankScreen: React.FunctionComponent = () => <View />;
+const SearchBankScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState(
+    [] as typeof props.bankList
+  );
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
 
-const mapStateToProps = (_: GlobalState) => ({});
+  const renderListItem = (isList: boolean) => (
+    info: ListRenderItemInfo<any>
+  ) => (
+    <BankPreviewItem
+      bank={info.item}
+      inList={isList}
+      onPress={props.onItemPress}
+    />
+  );
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBankScreen);
+  const handleFilter = (text: string) => {
+    setIsSearchStarted(true);
+    const resultList = props.bankList.filter(
+      bank =>
+        bank.abi.toLowerCase().indexOf(text.toLowerCase()) > -1 ||
+        bank.name.toLowerCase().indexOf(text.toLowerCase()) > -1
+    );
+    setFilteredList(resultList);
+    setSearchText(text);
+  };
+
+  const keyExtractor = (bank: any): string => bank.abi;
+
+  const openTosModal = () => {
+    props.showModal(
+      <TosBonusComponent
+        tos_url={"https://google.com"}
+        onClose={props.hideModal}
+      />
+    );
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack={true}
+      headerTitle={I18n.t("bonus.bpd.searchAbi.title")}
+    >
+      <SafeAreaView style={{ flex: 1 }}>
+        <Content style={{ flex: 1 }}>
+          {!isSearchStarted && (
+            <>
+              <View spacer={true} large={true} />
+              <Text>
+                <Text bold={true}>
+                  {I18n.t("bonus.bpd.searchAbi.description.text1")}
+                </Text>
+                {I18n.t("bonus.bpd.searchAbi.description.text2")}
+                <Link onPress={openTosModal}>
+                  {I18n.t("bonus.bpd.searchAbi.description.text3")}
+                </Link>
+              </Text>
+            </>
+          )}
+          <View spacer={true} />
+          <Label>{I18n.t("bonus.bpd.searchAbi.bankName")}</Label>
+          <Item>
+            <Input
+              placeholder={I18n.t("bonus.bpd.searchAbi.bankName")}
+              value={searchText}
+              onChangeText={handleFilter}
```

Comment:
To avoid wasting searches while the user is typing, you could use [debounce](https://lodash.com/docs/4.17.15#debounce) approach by settings (200-300milliseconds as delay)

Reply:
solved on 0cb9afb